### DISCUSSION TITLE
App is not running after turn off display

### DISCUSSION THREAD

**Discussion by beppo-ivel** [appcelerator/titanium_mobile#13639] [Q&A] [answered]
I deployed simple app to mobile. If I click button, then it writes the time every 10 seconds to a file. I open the App click the start button, put the app in background. Around `10:20` I turn off display and do nothing. Around `10:26` I turn on display and open the app to stop the interval. As you can see there the interval has gaps between `10:20:25` and `10:26:59`. Same if I implemented the interval as a service.  How to keep app running after turn off display? I expected a service is doing this by default. If I ggogle it, then the keyword `wakeLock` appears. is there a tutorial for titanium?

I can also explain the 3 records in the gap. The mobile display flashed on and off.

```
[10:18:55] interval ...
[10:19:05] interval ...
[10:19:15] interval ...
[10:19:25] interval ...
[10:19:35] interval ...
[10:19:45] interval ...
[10:19:55] interval ...
[10:20:05] interval ...
[10:20:15] interval ...
[10:20:25] interval ...
[10:23:16] interval ...
[10:24:31] interval ...
[10:26:41] interval ... 
[10:26:59] interval ... 
[10:27:09] interval ... 
[10:27:19] interval ... 
[10:27:29] interval ... 
```

```

function onClickStart(e) {
  f.write("");
  interval = setInterval(function() {
    let date = new Date();
    let dateFormat = '' + ("0" + date.getHours()).slice(-2) + ':' + ("0" + date.getMinutes()).slice(-2) + ':' + ("0" + date.getSeconds()).slice(-2);
    f.append("[" + dateFormat + "] interval ... \n");
  }, 1000*10);
}
```

**Comment by m1ga** [comment, marked as answer]
never used it, can just point you to the readme part:
https://titaniumsdk.com/api/titanium/android/notification.html#wakelock
and use one of
https://titaniumsdk.com/api/structs/wakelockoptions.html

and there is https://titaniumsdk.com/api/titanium/ui/window.html#properties_keepscreenon but don't think you want that but those are the options in the "keep the device awake" from Android https://developer.android.com/training/scheduling/wakelock

**Reply by beppo-ivel** [reply]
How to apply wakeLock options? Is there a small example.

**Reply by m1ga** [reply]
It's a property of the notification, so you put that key:value in createNotification of your foregroundNotification
